index.js

```diff
@@ -530,7 +530,15 @@ class ServerlessAmplifyPlugin {
         const resource = resources.find(r => r.ResourceType === 'AWS::AppSync::GraphQLApi');
         if (resource) {
             const schemaFile = this.getTemporarySchemaFile(resource);
-            graphqlGenerator(schemaFile, fileDetails.filename, { language: 'graphql' });
+            const fileType = path.extname(fileDetails.filename).substr(1);
+            const language = {
+              ts: 'typescript',
+              js: 'javascript',
+              swift: 'swift',
+              scala: 'scala',
+              graphql: 'graphql',
+            }[fileType];
+            graphqlGenerator(schemaFile, fileDetails.filename, { language });
```

Comment:
This does not provide any error handling when the language is not one of those listed.  In that case, language will be explicitly undefined and that will cause horrible errors.  Please add error handling appropriate to the situation.